File: back-end/mercado-inteligente-spring/src/main/java/br/edu/usf/mercadointeligentespring/core/mapper/UserMapper.java
package br.edu.usf.mercadointeligentespring.core.mapper;

import br.edu.usf.mercadointeligentespring.core.domain.User;
import br.edu.usf.mercadointeligentespring.gateways.dto.UserDTO;
import org.mapstruct.Mapper;
import org.mapstruct.ReportingPolicy;
import org.springframework.stereotype.Component;

@Component
@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.WARN)
public interface UserMapper {
    UserDTO toDTO(User user);

    User toDomain(UserDTO userDTO);
}

File: back-end/mercado-inteligente/src/main/kotlin/br/edu/usf/controllers/mapper/UserMapper.kt
package br.edu.usf.controllers.mapper

import br.edu.usf.core.domain.User
import br.edu.usf.gateways.dto.UserDTO
import org.mapstruct.Mapper

@Mapper
interface UserMapper {
    fun toDTO(user: User): UserDTO
    fun toDomain(user: UserDTO): User
}